src/xdg.rs

```diff
@@ -245,11 +266,14 @@ impl IconLoader {
     fn load(&self, icon: &str, size: u32) -> Result<Icon, Error> {
         let name = icon.into();
 
-        let path = self.icons.get(icon).ok_or(Error::NotFound)?;
-        let path_str = path.to_string_lossy();
+        // Resolve icon from name if it is not an absolute path.
+        let mut path = Path::new(icon);
+        if !path.is_absolute() {
+            path = self.icons.get(icon).ok_or(Error::NotFound)?;
+        }
 
-        match &path_str[path_str.len() - 4..] {
-            ".png" => {
+        match path.extension().and_then(|ext| ext.to_str()) {
+            Some("png") => {
```

Comment:
Checking for extension is a bit more clear about the intention of the code, less error-prone if we ever wanted to support `.whatever` and it shouldn't really impact performance since `to_string_lossy` is likely more expensive than `extension()`.